model.py

```diff
@@ -238,19 +240,18 @@ def from_json(cls, json_dict):
 
     @classmethod
     def from_pending_tournament(cls, pending_tournament):
+        # takes a real alias to id map instead of a list of objects
         def _get_player_id_from_map_or_throw(alias_to_id_map, alias):
-            for mapping in alias_to_id_map:
-                if mapping['player_alias'] == alias:
-                    player_id = mapping['player_id']
-                    if player_id is None:
-                        raise Exception('Alias %s has no ID in map\n: %s' % (alias, alias_to_id_map))
-                    else:
-                        return player_id
+            if alias in alias_to_id_map:
+                if alias_to_id_map[alias] is None:
+                    raise ValueError('Alias %s has no ID in map\n: %s' % (alias, alias_to_id_map))
+                else:
+                    return alias_to_id_map[alias]
```

Comment:
If you're converting alias_to_id_map into an actual map before passing it in, this function can be much simpler.